content/en/apps/guides/hosting/4.x/data-migration.md

```diff
@@ -0,0 +1,244 @@
+
+##### a. CouchDB secret 
+Used in encrypting all CouchDb passwords and session tokens.
+##### b. CouchDb server uuid
+Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
+
+Use the migration tool to obtain these values.
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration get-env
+```
+
+### 4. Locate and backup CouchDb Data folder
+a) If running in MedicOS, [CouchDb data folder]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}) can be found at `/srv/storage/medic-core/couchdb/data`.
+
+b) If running a custom installation of CouchDb, data would be typically stored at `/opt/couchdb/data`.
+
+c) TODO when using orchestration / AWS 
+
+### 5. Launch 4.x CouchDb installation
+
+Depending on your project scalability needs and technical possibilities, you must decide whether you will deploy CouchDb in a single node or in a cluster with multiple nodes.
+Please consult this guide about clustering and horizontal scalability to make an informed decision. <insert link>
+
+{{% alert title="Note" %}} You can start with single node and then change to a cluster. This involves running the migration tool again to distribute shards from the existent node to the new nodes. {{% /alert %}}
+
+Depending on your choice, follow the instructions that match your deployment below: 
+
+#### Single node
+
+a) Download 4.x single-node CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-single/ 
+cd ~/couchdb-single/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb.yml
+```
+a) Make a copy of the 3.x CouchDb data folder from **step 4**.
+
+b) Set the correct environment variables:
+```shell
+cat > ${HOME}/couchdb-single/.env << EOF
+COUCHDB_USER=<admin>
+COUCHDB_PASSWORD=<password>
+COUCHDB_SECRET=<COUCHDB_SECRET from step 3>
+COUCHDB_UUID=<COUCHDB_UUID from step 3>
+COUCHDB_DATA=<absolute path to folder created in step 5.a>
+EOF
+```
+c) Start 4.x CouchDb and wait until it is up.
+```shell
+cd ~/couchdb-single/ 
+docker-compose up -d
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration check-couchdb-up
+```
+
+d) Change metadata to match the new CouchDb node
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration move-node
+```
+e) Verify that the migration was successful
+```shell
+docker-compose run couch-migration verify
+```
+If all checks pass, proceed with starting CHT-Core 4.x, using the same environment variables.   
+
+#### Multi node
+
+a) Download 4.x clustered CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-cluster/ 
+cd ~/couchdb-cluster/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb-clustered.yml
+```
+b) Create a data folder for every one of the CouchDb nodes.
+
```

Comment:
Seems worth an example (though I had trouble formatting code inside a suggestion :disappointed: ) :

```
b) Create a data folder for every one of the CouchDb nodes. If you were going to a 3 cluster node, this would be:

   mkdir -p ~/couchdb-cluster/node1
   mkdir -p ~/couchdb-cluster/node2
   mkdir -p ~/couchdb-cluster/node3

```

(Ah - i see below you call it `couchdb_data_main` vs `node1`.  I like `main` as it's less ambiguous which is the main - which ever you want to standardize on!)